### PR TITLE
[TVMScript] Refactor IRDocsifier

### PR DESCRIPTION
**Pull request by junrushao**
This PR refactors the TVMScript printer and includes the following
changes:
- Consolidate the logics of VarTable into IRDocsifier
- Decouple TracedObject into Object and ObjectPath for less syntactic
  noise
- Restructure the folder to ensure logics and consistency

Some tests removed because the APIs do not exist any more due to the
consolidation.

### CODE REVIEW

File: include/tvm/script/printer/printer.h [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+namespace printer {
+
+/*! \brief Default values in the TVMScript printer */
+struct Default {
+  /*! \brief Default data type of TIR buffer */
+  DataType buffer_dtype = DataType::Float(32);
+  /*! \brief Default data type of integer literals */
+  DataType int_dtype = DataType::Int(32);
+  /*! \brief Default data type of float literals */
+  DataType float_dtype = DataType::Void();
```
Comment (Hzfengsy) [inline include/tvm/script/printer/printer.h:39]: Why is it `DataType::Void()`?

File: src/script/printer/doc_printer/python_doc_printer.cc [hunk 2]
```diff
@@ -294,7 +295,11 @@ void PythonDocPrinter::PrintTypedDoc(const LiteralDoc& doc) {
   } else if (const auto* float_imm = value.as<FloatImmNode>()) {
     // TODO(yelite): Make float number printing roundtrippable
```
Comment (Hzfengsy) [inline src/script/printer/doc_printer/python_doc_printer.cc:296]: Do you know if this TODO is resolved?

File: tests/scripts/task_mypy.sh [hunk 3]
```diff
@@ -32,9 +32,6 @@ mypy  --check-untyped-defs python/tvm/tir/analysis/
 echo "Checking MyPy Type defs in the transform package."
 mypy  --check-untyped-defs python/tvm/tir/transform/
 
-echo "Checking MyPy Type defs in the tvmscript printer package."
-mypy  --check-untyped-defs python/tvm/script/printer
```
Comment (Hzfengsy) [inline tests/scripts/task_mypy.sh:36]: Please enable mypy checks

**Review by Hzfengsy — APPROVED** [review]
Overall LGTM

File: tests/scripts/task_mypy.sh (hunk 3 above)
Comment (junrushao) [reply tests/scripts/task_mypy.sh:36]: sure!

File: include/tvm/script/printer/printer.h (hunk 1 above)
Comment (junrushao) [reply include/tvm/script/printer/printer.h:39]: I looked up the current convention and it seems that we always print T.float32/float64 for floating point numbers, so this is to be consistent with the current style

File: src/script/printer/doc_printer/python_doc_printer.cc (hunk 2 above)
Comment (junrushao) [reply src/script/printer/doc_printer/python_doc_printer.cc:296]: Not yet but indeed there is a way forward

File: include/tvm/script/printer/printer.h (hunk 1 above)
Comment (Hzfengsy) [reply include/tvm/script/printer/printer.h:39]: Please remove it if it is not used. All at least add a comment for better readability.

Comment (junrushao) [reply include/tvm/script/printer/printer.h:39]: I do think that we could consider a default dtype for float in the near future

Comment (tqchen) [reply include/tvm/script/printer/printer.h:39]: I think we can change the comment to "Right now we always print out the explicit type of floating point values, so setting it to Void means we do not print without T.float32/T.float64 wrapper"

Comment (junrushao) [reply include/tvm/script/printer/printer.h:39]: that sounds good!

**Review by tqchen — APPROVED** [review]

**Comment by tqchen** [comment]
Thanks @junrushao @Hzfengsy !

**Pull request merged**